src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -107,18 +121,27 @@ def _extract_messages(self) -> None:
                     exc_info=True,
                 )
                 continue
-
             try:
                 if self.__input_codec:
                     self.__input_codec.validate(parsed_payload)
             except ValidationError:
                 # For now while this is still experimental, those errors are
                 # not supposed to be fatal.
-                logger.warn(
+                logger.warning(
                     "process_messages.invalid_schema",
                     extra={"payload_value": str(msg.payload.value)},
                     exc_info=True,
                 )
+            if use_case_id := extract_use_case_id(parsed_payload["name"]):
+                parsed_payload["use_case_id"] = use_case_id
+            else:
+                logger.error(
+                    "process_messages.invalid_metric_resource_identifier",
+                    extra={"payload_value": str(msg.payload.value)},
+                    exc_info=True,
+                )
+                continue
```

Comment:
What does this `continue` do?

Reply:
It stops [line 144](https://github.com/getsentry/sentry/blob/f2cc46ed4e49d48280d3e8a6dca7e67924cd5b3f/src/sentry/sentry_metrics/consumers/indexer/batch.py#L144) from being run because use_case_id is not valid.

I should add 
```python
self.skipped_offsets.add(partition_offset)
```
before the `logger.error`

Reply:
Oh right... just a nit but i think it's more readable if you just move line 144 up into the if block and dropping this

Reply:
I see, I was just repeating the pattern from above [(line 114 ~ 123)](https://github.com/getsentry/sentry/blob/f2cc46ed4e49d48280d3e8a6dca7e67924cd5b3f/src/sentry/sentry_metrics/consumers/indexer/batch.py#L114-L123)
```python
try:
    parsed_payload = json.loads(msg.payload.value.decode("utf-8"), use_rapid_json=True)
except rapidjson.JSONDecodeError:
    self.skipped_offsets.add(partition_offset)
    logger.error(
        "process_messages.invalid_json",
        extra={"payload_value": str(msg.payload.value)},
        exc_info=True,
    )
    continue
>>>> My code
if use_case_id := extract_use_case_id(parsed_payload["name"]):
    parsed_payload["use_case_id"] = use_case_id
else:
    logger.error(
        "process_messages.invalid_metric_resource_identifier",
        extra={"payload_value": str(msg.payload.value)},
        exc_info=True,
    )
    continue
<<<<
self.parsed_payloads_by_offset[partition_offset] = parsed_payload // Skipped by both continue keywords
```
I feel like nestling `self.parsed_payloads_by_offset[partition_offset] = parsed_payload` into the if statement might be a little confusing because it will no longer be apparent what the `try` / `except` statements' `continue` are skipping.